src/routes/user.route.js

```diff
@@ -0,0 +1,63 @@
+const express = require("express");
+const router = express.Router();
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+const {User} = require('../models/user.model')
+const {UserData} = require('../models/userData.model')
+
+const saltRounds = 10;
+const secret =
+  "gEf3gs2kMagd2CHDXpEjwGJlbVewlqE7ARhD2UIYUJsM8V9c71E4rYGI0AXIn5J2pAi2TrMQHD7FAJOTtLIYIShHDjZjTwvzHodjwutvbTcFCnksFiCBFpqpQqucxyairN4X3hQbmiuKqJQW9XSeblgU5v2BQrEZs86YhgrGxQtWujl3NMu9NRwa9x9noEY7OnuabuIAYNCmsnGC39iq32nBOPvtA4BP+sVuMKW6Qd6//lSr1pSzYHZ9KMYUDnTDvnXQ3q5uSpCGcRTS3//G/nfUdjxjcbz1z9B2hvL+Oh/RXNI1DeFdwphwP9pbJqICm81l1WUe0H7Vs/6irJEq6w==";
+
+router
+  .route("/signup")
+  // Signup
+  .post(async (req, res) => {
+    try {
+      const user = req.body;
+      console.log(req.body)
+      bcrypt.hash(user.password, saltRounds, async function (err, hash) {
+        const NewUser = new User({ ...user, password: hash });
+        await NewUser.save();
+        const NewUserData = new UserData({ userId: NewUser._id });
+        await NewUserData.save();
+        return res.json({
+          status: "Signup successful",
+          user: NewUser,
+          userData: NewUserData,
+        });
+      });
+    } catch (error) {
+      res
+        .status(404)
```

Comment:
I think for authentication failure we use 401 instead of 404
```suggestion
        .status(401)
```